pkg/tsdb/azuremonitor/loganalytics/azure-log-analytics-datasource.go

```diff
@@ -229,14 +230,22 @@ func appendErrorNotice(frame *data.Frame, err *AzureLogAnalyticsAPIError) {
 	}
 }
 
-func (e *AzureLogAnalyticsDatasource) createRequest(ctx context.Context, dsInfo types.DatasourceInfo, url string) (*http.Request, error) {
+func (e *AzureLogAnalyticsDatasource) createRequest(ctx context.Context, dsInfo types.DatasourceInfo, url string, originalHTTPHeaders map[string]string) (*http.Request, error) {
```

Comment:
this function is starting to grow and is duplicated in 3 places. Can you move it to a `utils` (or `http`) package so it can be reused?